formatters.go

```diff
@@ -2,43 +2,51 @@ package main
 
 import (
 	"fmt"
+	"net/http"
 	"sort"
 	"strings"
 
 	"github.com/crowdsecurity/crowdsec/pkg/models"
 )
 
-var FormattersByName map[string]func([]*models.Decision) string = map[string]func([]*models.Decision) string{
+var FormattersByName map[string]func(w http.ResponseWriter, r *http.Request) = map[string]func(w http.ResponseWriter, r *http.Request){
 	"plain_text": PlainTextFormatter,
 	"microtik":   MicroTikFormatter,
 }
 
-func PlainTextFormatter(decisions []*models.Decision) string {
+func PlainTextFormatter(w http.ResponseWriter, r *http.Request) {
+	decisions := r.Context().Value(globalDecisionRegistry.Key).([]*models.Decision)
 	ips := make([]string, len(decisions))
 	for i, decision := range decisions {
 		ips[i] = *decision.Value
 	}
 	sort.Strings(ips)
-	return strings.Join(ips, "\n")
+	w.Write([]byte(strings.Join(ips, "\n")))
 }
 
-func MicroTikFormatter(decisions []*models.Decision) string {
+func MicroTikFormatter(w http.ResponseWriter, r *http.Request) {
+	decisions := r.Context().Value(globalDecisionRegistry.Key).([]*models.Decision)
 	ips := make([]string, len(decisions))
+	listName := r.URL.Query().Get("listname")
+	if listName == "" {
+		listName = "CrowdSec"
+	}
 	for i, decision := range decisions {
 		var ipType = "/ip"
 		if strings.Contains(*decision.Value, ":") {
 			ipType = "/ipv6"
 		}
 		ips[i] = fmt.Sprintf(
-			"%s firewall address-list add list=CrowdSec address=%s comment=\"%s for %s\"",
+			"%s firewall address-list add list=%s address=%s comment=\"%s for %s\"",
 			ipType,
+			listName,
 			*decision.Value,
 			*decision.Scenario,
 			*decision.Duration,
 		)
 	}
 	sort.Strings(ips)
-	return "/ip firewall address-list remove [find list=CrowdSec]\n" +
-		"/ipv6 firewall address-list remove [find list=CrowdSec]\n" +
-		strings.Join(ips, "\n")
+	w.Write([]byte(fmt.Sprintf("/ip firewall address-list remove [find list=%s]\n", listName) +
+		fmt.Sprintf("/ipv6 firewall address-list remove [find list=%s]\n", listName) +
```

Comment:
You need to remove this ipv6 line if it's v4 only mode because it may indicate that the user does not have v6 module enabled

Reply:
I have made it to remove v4 if onlyipv6 and remove v6 if only ipv4?

Reply:
Well, yes. But technically ipv4 is always enabled. But it would be cleaner